sql/src/main/java/org/apache/druid/sql/calcite/external/TableAppendMacro.java

```diff
@@ -0,0 +1,344 @@
+ */
+public class TableAppendMacro extends SqlUserDefinedTableMacro implements AuthorizableOperator
+{
+
+  public static final OperatorConversion OPERATOR_CONVERSION = new OperatorConversion();
+  public static final SqlOperator APPEND_TABLE_MACRO = new TableAppendMacro();
+
+  private static class OperatorConversion implements SqlOperatorConversion
+  {
+    public static final String FUNCTION_NAME = "APPEND";
+
+    public OperatorConversion()
+    {
+    }
+
+    @Override
+    public SqlOperator calciteOperator()
+    {
+      return APPEND_TABLE_MACRO;
+    }
+
+    @Override
+    public DruidExpression toDruidExpression(PlannerContext plannerContext, RowSignature rowSignature, RexNode rexNode)
+    {
+      throw new IllegalStateException();
+    }
+  }
+
+  private TableAppendMacro()
+  {
+    super(
+        new SqlIdentifier(OperatorConversion.FUNCTION_NAME, SqlParserPos.ZERO),
+        SqlKind.OTHER_FUNCTION,
+        ReturnTypes.CURSOR,
+        null,
+        new OperandMetadata(),
+        null
+    );
+  }
+
+  private static class OperandMetadata implements SqlOperandMetadata
+  {
+    @Override
+    public boolean checkOperandTypes(SqlCallBinding callBinding, boolean throwOnFailure)
+    {
+      for (int i = 0; i < callBinding.getOperandCount(); i++) {
+        SqlNode operand = callBinding.operand(i);
+        if (!callBinding.isOperandLiteral(i, false)) {
+          if (throwOnFailure) {
+            throw DruidSqlValidator.buildCalciteContextException(
+                "All arguments to APPEND should be literal strings. "
+                    + "Argument #" + i + " is not literal",
+                operand
+            );
+          } else {
+            return false;
+          }
+        }
+
+        SqlTypeName typeName = callBinding.getOperandType(i).getSqlTypeName();
+        if (!SqlTypeFamily.CHARACTER.getTypeNames().contains(typeName)) {
+          if (throwOnFailure) {
+            throw DruidSqlValidator.buildCalciteContextException(
+                "All arguments to APPEND should be literal strings. "
+                    + "Argument #" + i + " is not string",
+                operand
+            );
+          } else {
+            return false;
+          }
+        }
+      }
+      return true;
+    }
+
+    @Override
+    public SqlOperandCountRange getOperandCountRange()
+    {
+      return SqlOperandCountRanges.from(1);
+    }
+
+    @Override
+    public String getAllowedSignatures(SqlOperator op, String opName)
+    {
+      return "APPEND( <TABLE_NAME>[, <TABLE_NAME> ...] )";
+    }
+
+    @Override
+    public List<RelDataType> paramTypes(RelDataTypeFactory typeFactory)
+    {
+      RelDataType t = typeFactory.createSqlType(SqlTypeName.VARCHAR);
+      return ImmutableList.<RelDataType>builder().add(t, t).build();
+    }
+
+    @Override
+    public List<String> paramNames()
+    {
+      return ImmutableList.<String>builder().add("tableName", "tableName").build();
```

Comment:
should this generate a list of all the parameters with an independent name for each of them?